internal/internal_task_handlers.go

```diff
@@ -1197,8 +1203,16 @@ func skipDeterministicCheckForEvent(e *historypb.HistoryEvent) bool {
 		if markerName == versionMarkerName || markerName == mutableSideEffectMarkerName {
 			return true
 		}
-		// case enumspb.EVENT_TYPE_WORKFLOW_TASK_STARTED:
-		//	return true
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_COMPLETED:
```

Comment:
These result of a workflow execution is handled after we check the history so these will never be in the replayed history. The replayer already checks for for completed and continue as new outside the state machine.

Reply:
Yeah we would never check these before. 

>How does "activity task timed out" play into this

Oh that shouldn't, should be workflow task time out